src/d3a/d3a_core/simulation.py

```diff
@@ -362,13 +364,18 @@ def _execute_simulation(self, slot_resume, tick_resume, console=None):
 
                 current_tick_in_slot = tick_no % config.ticks_per_slot
                 if self.simulation_config.external_connection_enabled:
-                    self.redis_connection.publish_event_tick_myco()
                     if external_global_statistics.is_it_time_for_external_tick(
                             current_tick_in_slot):
                         external_global_statistics.update()
 
                 self.area.tick_and_dispatch()
                 self.area.update_area_current_tick()
+                if self.simulation_config.external_connection_enabled and \
+                        ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE == \
+                        d3a.constants.BidOfferMatchAlgoEnum.CUSTOM.value and \
```

Comment:
This is getting kind of long and I suggest to refactor this into a function (can be reused in L398 too)